python/cuspatial/cuspatial/core/binpreds/feature_equals.py

```diff
@@ -0,0 +1,342 @@
+            Coordinates sorted according to the bins defined by offsets.
+        """
+        result = self._sort_interleaved_points_by_offset(
+            coords, offsets, ["object_key", "xy", "xy_key"]
+        )
+        result.name = None
+        return result
+
+    def _sort_multipoints(self, lhs, rhs):
+        """Sort the coordinates of the multipoints in the left-hand and
+        right-hand GeoSeries.
+
+        Parameters
+        ----------
+        lhs : GeoSeries
+            The left-hand GeoSeries.
+        rhs : GeoSeries
+            The right-hand GeoSeries.
+
+        Returns
+        -------
+        lhs_result : Tuple
+            A tuple containing the sorted left-hand GeoSeries and the
+            sorted right-hand GeoSeries.
+        """
+        lhs_sorted = self._sort_multipoint_series(
+            lhs.multipoints.xy, lhs.multipoints.geometry_offset
+        )
+        rhs_sorted = self._sort_multipoint_series(
+            rhs.multipoints.xy, rhs.multipoints.geometry_offset
+        )
+        lhs_result = cuspatial.core.geoseries.GeoSeries.from_multipoints_xy(
+            lhs_sorted, lhs.multipoints.geometry_offset
+        )
+        rhs_result = cuspatial.core.geoseries.GeoSeries.from_multipoints_xy(
+            rhs_sorted, rhs.multipoints.geometry_offset
+        )
+        lhs_result.index = lhs.index
+        rhs_result.index = rhs.index
+        return (
+            lhs_result,
+            rhs_result,
+        )
+
+    def _reverse_linestrings(self, coords, offsets):
+        """Reverse the order of coordinates in a Arrow buffer of coordinates
+        and offsets."""
+        result = self._sort_interleaved_points_by_offset(
+            coords, offsets, ["object_key", "point_key", "xy_key"]
+        )
+        result.name = None
+        return result
+
+    def _preprocess(self, lhs: "GeoSeries", rhs: "GeoSeries"):
+        """Convert the input geometry types into buffers of points that can
+        then be compared with the equals basic predicate.
+
+        The equals basic predicate is simply the value-equality operation
+        on the coordinates of the points in the geometries. This means that
+        we can convert any geometry type into a buffer of points and then
+        compare the buffers with the equals basic predicate.
+
+        Parameters
+        ----------
+        lhs : GeoSeries
+            The left-hand GeoSeries.
+        rhs : GeoSeries
+            The right-hand GeoSeries.
+
+        Returns
+        -------
+        result : GeoSeries
+            A GeoSeries of boolean values indicating whether each feature in
+            the right-hand GeoSeries satisfies the requirements of a binary
+            predicate with its corresponding feature in the left-hand
+            GeoSeries.
+        """
+        type_compare = lhs.feature_types == rhs.feature_types
+        # Any unmatched type is not equal
+        if (type_compare == False).all():  # noqa: E712
```

Comment:
Could simplify and get rid of the `noqa`.

```suggestion
        # Any unmatched type is not equal
        if (lhs.feature_types != rhs.feature_types).all():
```

Reply:
I like that improvement, done.